src/common/pool_map.c

```diff
@@ -91,6 +91,11 @@ struct pool_map {
 	 * NB: All components must be stored in contiguous buffer.
 	 */
 	struct pool_domain	*po_tree;
+	/**
+	 * number of currently failed pool components of each type
+	 * of component found in the pool
+	 */
+	struct pool_fail_comp	*po_comp_fail_cnts;
```

Comment:
I don't quite like to introduce this in pool map, it's actually only being updated & used on pool leader, keep it on every server seems redundant and confusing to me. Why don't we just scan the pool_map to get fail count on cont_open()? That could simplify the code IMHO.

Reply:
I think this could be helpful for future work because we do want to fail object open which could be more performance sensitive, if there is a cache result like this, then it would be much cheaper?

Reply:
So, it will be updated on each server and client? (I see it's only updated on pool leader in this patch)

Reply:
I missed the check for the pool leader at the top of ds_pool_update_internal().
In that case should it instead go in ds_pool_tgt_map_update()?

Reply:
Right, if you want it being used on other servers, you'd move it into ds_pool_tgt_map_update(), if you want it being used on client, you need to update it in pool_map_initialise()? as well.

Reply:
When I talked to Liang about it last it sounded like it only needed to be updated and used server side.
@gnailzenh Is that correct?

If it would be useful client side as well I can just add the update call there as well.

Reply:
yep, I think this could be used by both servers and clients, we can fix this follow-on patch